File: ocean-core/src/main/java/org/matrixstudio/ocean/core/util/JacksonMapper.java
package org.matrixstudio.ocean.core.util;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.IOException;

/**
 * <p>fasterxml jackson mapper utils.</p>
 *
 * @author liuwei
 */
public class JacksonMapper {

    private static final Logger LOGGER = LoggerFactory.getLogger(JacksonMapper.class);

    /**
     * {@link ObjectMapper}, initialize by constructor method
     */
    private ObjectMapper objectMapper;

    /**
     * <p>Initialize, must provide a {@link ObjectMapper} object.</p>
     *
     * @param objectMapper a {@link ObjectMapper} object
     */
    public JacksonMapper(ObjectMapper objectMapper) {
        this.objectMapper = objectMapper;
    }

    public ObjectMapper getObjectMapper() {
        return objectMapper;
    }

    public <T> T read(String jsonStr, Class<T> clazz) {
        T object = null;
        try {
            object = objectMapper.readValue(jsonStr, clazz);
        } catch (IOException e) {
            LOGGER.error(e.getMessage(), e);
        }
        return object;
    }

    public <T> T read(String jsonStr, TypeReference<T> typeReference) {
        T object = null;
        try {
            object = objectMapper.readValue(jsonStr, typeReference);
        } catch (IOException e) {
            LOGGER.error(e.getMessage(), e);
        }
        return object;
    }

    public String write(Object object) {
        String jsonStr = null;
        try {
            jsonStr = objectMapper.writeValueAsString(object);
        } catch (JsonProcessingException e) {
            LOGGER.error(e.getMessage(), e);
        }
        return jsonStr;
    }
}

File: ocean-core/src/main/kotlin/io/lw900925/ocean/core/util/JacksonMapper.kt
package io.lw900925.ocean.core.util

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class JacksonMapper(var objectMapper: ObjectMapper) {

    val LOGGER: Logger = LoggerFactory.getLogger(JacksonMapper::class.java)

    fun <T> read(jsonStr: String, clazz: Class<T>): T? {
        return objectMapper.readValue(jsonStr, clazz)
    }

    fun <T> read(jsonStr: String, typeReference: TypeReference<T>): T? {
        return objectMapper.readValue(jsonStr, typeReference)
    }

    fun write(`object`: Any): String {
        return objectMapper.writeValueAsString(`object`)
    }
}